internal/ingress/controller/parser/parser.go

```diff
@@ -357,10 +368,10 @@ func overrideRoute(route *Route,
 	if r.RegexPriority != nil {
 		route.RegexPriority = kong.Int(*r.RegexPriority)
 	}
-	if r.StripPath != nil {
+	if r.StripPath != nil && !isTLSNegotiation {
 		route.StripPath = kong.Bool(*r.StripPath)
 	}
-	if r.PreserveHost != nil {
+	if r.PreserveHost != nil && !isTLSNegotiation {
```

Comment:
Let's leave `overrirdeRoute` method as is for now.
In case there is a use case where the user actually wants to set these properties using a configuration.konghq.com annotation, we should allow that. The current change would result in a situation where a user can never change these properties.